webapp/src/ts/services/fast-action-button.service.ts

```diff
@@ -0,0 +1,236 @@
+import { Inject, Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { DOCUMENT } from '@angular/common';
+import { Store } from '@ngrx/store';
+
+import { AuthService } from '@mm-services/auth.service';
+import { ResponsiveService } from '@mm-services/responsive.service';
+import { ReportsActions } from '@mm-actions/reports';
+import { ButtonType } from '@mm-components/fast-action-button/fast-action-button.component';
+import { TranslateService } from '@mm-services/translate.service';
+import { TranslateFromService } from '@mm-services/translate-from.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class FastActionButtonService {
+
+  private reportsActions: ReportsActions;
+
+  constructor(
+    private router: Router,
+    private store: Store,
+    private authService: AuthService,
+    private responsiveService: ResponsiveService,
+    private translateService: TranslateService,
+    private translateFromService: TranslateFromService,
+    @Inject(DOCUMENT) private document: Document,
+  ) {
+    this.reportsActions = new ReportsActions(store);
+  }
+
+  private executeMailto(link) {
+    const element = this.document.createElement('a');
+    element.href = link;
+    element.click();
+    element.remove();
+  }
```

Comment:
I _think_ we should get the same results with a simple `window.open(link)` but I have some doubts if that would work within the cht-android wrapper

Reply:
I first implemented this using `window.open(link)` but it doesn't open the mailto for all cases, I can't remember right now if it was because of the Angular navigation in some pages or because of the cht-android. But creating the element and do click, was the most reliable way I found back then to trigger the mailto from JS code.